File: src/main/java/javabot/commands/EnableOperation.java
package javabot.commands;

import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

import javabot.Message;
import javabot.Javabot;
import javabot.BotEvent;

/**
 * Created Jan 26, 2009
 *
 * @author <a href="mailto:jlee@antwerkz.com">Justin Lee</a>
 */
public class EnableOperation extends OperationsCommand {
    @Override
    public List<Message> execute(final Javabot bot, final BotEvent event, final List<String> args) {
        final List<Message> messages = new ArrayList<Message>();
        if (args.isEmpty()) {
            bot.postMessage(new Message(event.getChannel(), event, "usage: enableOperation <name>"));
            bot.postMessage(new Message(event.getChannel(), event,
                "usage: use admin listOperations to see list of options"));
        } else {
            final String name = args.get(0);
            if(bot.addOperation(name)) {
                bot.postMessage(new Message(event.getChannel(), event, name + " successfully enabled."));
                listCurrent(bot, event);
            } else {
                bot.postMessage(new Message(event.getChannel(), event, name + " not enabled.  Either it is already running"
                    + " or it's not a valid name.  see listOperations for details."));
            }
        }
        return Collections.<Message>emptyList();
    }
}

File: src/main/kotlin/javabot/commands/EnableOperation.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import com.google.inject.Inject
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

class EnableOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) : OperationsCommand(bot, adminDao) {
    @Parameter(names = arrayOf("--name"))
    lateinit var operationName: String

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        bot.enableOperation(operationName)
        responses.add(Message(event, Sofia.adminOperationEnabled(operationName)))
        listCurrent(event, responses)
        
        return responses
    }
}
